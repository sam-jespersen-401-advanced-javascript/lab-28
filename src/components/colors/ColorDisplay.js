import React from 'react';
import PropTypes from 'prop-types';

const ColorDisplay = ({ backgroundColor }) => (
  <div style={{
    backgroundColor,
    width: '100px',
    height: '100px'
  }}></div>
);

ColorDisplay.propTypes = {
  backgroundColor: PropTypes.string.isRequired
};

export default ColorDisplay;
