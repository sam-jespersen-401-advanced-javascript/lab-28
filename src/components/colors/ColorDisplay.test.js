import React from 'react';
import { shallow } from 'enzyme';
import ColorDisplay from './ColorDisplay';

describe('ColorDisplay module', () => {
  it('renders a colorDisplay', () => {
    const wrapper = shallow(<ColorDisplay backgroundColor="red" />);
    expect(wrapper).toMatchSnapshot();
  });
});
