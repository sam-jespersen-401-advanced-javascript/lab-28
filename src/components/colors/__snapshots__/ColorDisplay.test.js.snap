// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorDisplay module renders a colorDisplay 1`] = `
<div
  style={
    Object {
      "backgroundColor": "red",
      "backgroundImage": undefined,
      "backgroundSize": "cover",
      "height": "100px",
      "width": "100px",
    }
  }
/>
`;
